packages/rgbpp/src/rgbpp/types.ts

```diff
@@ -0,0 +1,25 @@
+import { DataSource } from '@rgbpp-sdk/btc';
+import { BtcTransferVirtualTxResult, Collector, Hex } from '@rgbpp-sdk/ckb';
+
+export interface RgbppTransferTxParams {
+  // The collector that collects CKB live cells and transactions
+  collector: Collector;
+  // The transferred RGB++ xUDT type script args
+  xudtTypeArgs: Hex;
+  // The rgbpp assets cell lock script args array whose data structure is: out_index | bitcoin_tx_id
+  rgbppLockArgsList: Hex[];
+  // The XUDT amount to be transferred, if the noMergeOutputCells is true, the transferAmount will be ignored
+  transferAmount: bigint;
+  // The sender BTC address
+  fromBtcAddress: string;
+  // The receiver BTC address
+  toBtcAddress: string;
+  btcDataSource: DataSource;
```

Comment:
Random idea out of nowhere—this is not a change request, just a topic of discussion: how about separating the props of BTC/CKB for better readability and easier prop management? For example:
```ts
{
  ckb: {
    collector: ...,
    xudtTypeArgs: ...,
    ...
  },
  btc: {
    from: ...,
    fromPubkey: ...,
    to: ...,
    ...
  },
}
```

Reply:
Maybe it's a good idea.

Reply:
Updated in https://github.com/ckb-cell/rgbpp-sdk/pull/216/commits/7bd4e7f40236d7d804d4a4085e85a5b6ed38c0ad